src/RequestManager.ts

```diff
@@ -43,11 +44,11 @@ class RequestManager {
     return this.transports[0];
   }
 
-  public async request(method: string, params: any[], notification: boolean = false, timeout?: number): Promise<any> {
+  public async request(requestObject: RequestArguments, notification: boolean = false, timeout?: number): Promise<any> {
```

Comment:
In theory thew new Message type should be the type here, I think it might read better